src/storage/sql/mod.rs

```diff
@@ -173,16 +174,37 @@ impl Storage for SqlStorage {
 
                 query = query.filter(snippets::id.eq_any(snippet_ids));
             }
-            if let Some(limit) = criteria.limit {
-                query = query.limit(limit as i64);
-            }
-            if let Some(offset) = criteria.offset {
-                query = query.offset(offset as i64);
+
+            // pagination
+            if let Some(marker) = criteria.pagination.marker {
+                let marker = self.get(&marker)?;
+                if let Some(created_at) = marker.created_at {
+                    query = match criteria.pagination.direction {
+                        Direction::Backward => query
+                            .filter(
+                                snippets::created_at
+                                    .le(chrono::DateTime::from(created_at))
```

Comment:
Why do we need `chrono::DateTime::from(created_at)`? I thought `created_at` is already an instance of `chrono:DataTime`.

Reply:
That's because of the wrapper that I previously added in #109 :( Previously, we used `chrono:DateTime` in both `Snippet` (the API model) and `Snippet` (the Diesel-level struct). Because we want the former to be JSON-serializable, I had to create a wrapper-type for `chrono:DateTime` (for which I could implement `Serializable`) and then use it in `Snippet` instead. I did not change the Diesel struct, though, so we still need to call from() when talking to Diesel.

Reply:
This makes me wonder if I should revert that and just implement `Serializable` for `Snippet` manually (which is fairly straightforward; it's just annoying that we can't derive it because of `chrono:DateTime` attributes)

Reply:
Let me know what you think about https://github.com/xsnippet/xsnippet-api/pull/114 . It's a bit less code :) While it's annoying that we have to implement `Serializable` explicitly, what I like about it is that we remove these redundant type conversion by the means of `into()` and `from()` that were scattered through the production code and tests.